csrc/ops/indexing.cpp

```diff
@@ -74,6 +74,10 @@ TensorView* indexSelect(
         "index_tv must be a 1d or scalar tensor.");
     index_tv =
         ops::maybeBroadcastIndexTv(index_tv->as<TensorView>(), dim, n_dims);
+  } else {
+    // We do an explicit set here, which would allow vectorized load on index_tv
+    // when applicable.
+    index_tv = set(index_tv->as<TensorView>());
```

Comment:
Why is this necessary? `cacheAfter` doesn't insert a cache for index tensors?

Reply:
hmmm. that's a good question. `cacheAfter` should have done that already for index_tv, (it skips the lookup_tv as intended).

Let me double check what went wrong.

Reply:
Good catch.
The cacheAfter was working properly. Followed through and we do have vectorized load without the extra set.

I probably overlooked this when I was dealing with the test check when validating vectorization factor for index_tv load.